src/libcore/macros.rs

```diff
@@ -20,6 +21,30 @@ macro_rules! panic {
     });
 }
 
+/// Panics the current thread.
+///
+/// For details, see `std::macros`.
+#[cfg(not(bootstrap))]
+#[macro_export]
+#[allow_internal_unstable(core_panic, panic_internals)]
+#[stable(feature = "core", since = "1.6.0")]
+macro_rules! panic {
+    () => (
+        $crate::panic!("explicit panic")
+    );
+    ($msg:expr) => ({
+        let loc = $crate::panic::Location::internal_constructor(file!(), line!(), column!());
+        $crate::panicking::panic($msg, &loc)
+    });
+    ($msg:expr,) => (
+        $crate::panic!($msg)
+    );
+    ($fmt:expr, $($arg:tt)+) => ({
+        let loc = $crate::panic::Location::internal_constructor(file!(), line!(), column!());
```

Comment:
One suggestion I had about this location was to use:
```rust
const LOC: &$crate::panic::Location = &$crate::panic::Location::internal_constructor(...);
```
to avoid codegen-ing a runtime call to the constructor and having a `Location` on a stack, effectively ensuring that codegen will be identical to before this PR and not cause any second-order effects etc.